flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/enumerator/initializer/SpecifiedOffsetsInitializer.java

```diff
@@ -38,6 +38,10 @@
  * An implementation of {@link OffsetsInitializer} which initializes the offsets of the partition
  * according to the user specified offsets.
  *
+ * <p>Use specified offsets for specified partitions while use commit offsets or earliest for
```

Comment:
@loserwang1024 I have a question.
When you mention `specified offsets`, do you mean the ones provided in the map `initialOffsets`.
If that is true, these unspecified partitions are initialized to be consumed according to `offsetResetStrategy`, which can be `earliest` or `latest`. So, not necessarily `earliest`

Also, could you point me to the code (or code change) that explains this
> Specified partition's offset should be less than its latest offset,
  otherwise it will start from the earliest.

Reply:
> These unspecified partitions are initialized to be consumed according to offsetResetStrategy
> 

yes, howerver, OffsetResetStrategy is set to earliest in OffsetsInitializer#offsets and `KafkaDynamicSource` .

Maybe I shoud give more desciptions like “use offsetResetStrategy which is default earliest”.

Yes, however, the OffsetResetStrategy is set to earliest in both OffsetsInitializer#offsets and KafkaDynamicSource. **Maybe I should add more details such as "use the default earliest offsetResetStrategy"**  .

> Specified partition's offset should be less than its latest offset, otherwise it will start from the earliest(maybe offsetResetStrategy).
> 

this code is in org.apache.kafka.clients.consumer.internals.Fetcher#handleOffsetOutOfRange, which use offsetResetStrategy for offsetOutOfRange.

You can see the last section of FLIP-288, where I provide a detailed explanation.

Reply:
@RamanVerma 
> These unspecified partitions are initialized to be consumed according to offsetResetStrategy
> 

yes, howerver, OffsetResetStrategy is set to earliest in OffsetsInitializer#offsets and `KafkaDynamicSource` .

Maybe I shoud give more desciptions like “use offsetResetStrategy which is default earliest”.

Yes, however, the OffsetResetStrategy is set to earliest in both OffsetsInitializer#offsets and KafkaDynamicSource. **Maybe I should add more details such as "use the default earliest offsetResetStrategy"**  .

> Specified partition's offset should be less than its latest offset, otherwise it will start from the earliest(maybe offsetResetStrategy).
> 

this code is in org.apache.kafka.clients.consumer.internals.Fetcher#handleOffsetOutOfRange, which use offsetResetStrategy for offsetOutOfRange.

You can see the last section of FLIP-288, where I provide a detailed explanation.

Reply:
thanks for the explanation @loserwang1024 
As you said, you can change that comment to "use offsetResetStrategy which is default earliest”.